tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,219 @@
+"""
+Name: post_copy_request_to_queue.py
+Description:  lambda function that queries the db for file metadata, updates the status
+of recovered file to staged, and sends the staged file info to staged_recovery queue for further processing.
+
+"""
+from enum import Enum
+import json
+import os
+import boto3
+from typing import Dict, Any, List
+import time
+import random
+from orca_shared import shared_recovery
+import requests_db
+import database
+from cumulus_logger import CumulusLogger
+import logging
+
+# instantiate CumulusLogger
+LOGGER = CumulusLogger()
+
+
+def task(
+    records: Dict[str, Any],
+    db_queue_url: str,
+    recovery_queue_url: str,
+    max_retries: int,
+    retry_sleep_secs: int,
+    retry_backoff: int,
+) -> None:
+    """
+    Task called by the handler to perform the work.
+
+    This task queries all entries from orca_recoverfile table
+    that match the given filename and whose status_id is 'PENDING'.
+    The result is then sent to the staged-recovery-queue SQS and status-update-queue SQS.
+
+    Args:
+        records: A dictionary passed through from the handler.
+        db_queue_url: The SQS URL of status_update_queue
+        recovery_queue_url: The SQS URL of staged_recovery_queue
+        max_retries: Number of times the code will retry in case of failure.
+        retry_sleep_secs: Number of seconds to wait between recovery failure retries.
+        retry_backoff: The multiplier by which the retry interval increases during each attempt.
+    Returns:
+        None
+    Raises:
+        Exception: If unable to retrieve key_path or db parameters, convert db result to json,
+        or post to queue.
+
+    """
+    # grab the key_path and bucketname from event
+    # We only expect one record.
+    key_path = records["s3"]["object"]["key"]
+    bucket_name = records["s3"]["bucket"]["name"]
+
+    sql = """
+        SELECT
+            job_id, granule_id, filename, restore_destination
+        FROM
+            orca_recoverfile
+        WHERE
+            key_path = %s
+        AND
+            status_id = %d
+    """
+    # Gets the dbconnection info.
+    # query the db table
+    try:
+        db_connect_info = requests_db.get_dbconnect_info()
+        rows = database.single_query(sql, db_connect_info, (key_path, shared_recovery.OrcaStatus.PENDING.value))
+        if len(rows) == 0:
+            message = f"No metadata found for {key_path}"
+            LOGGER.fatal(message)
+            raise Exception(message)
+        # # convert db result to json
+        db_result_json = database.result_to_json(rows)
+
+    except Exception:
```

Comment:
Should probably log the exception. In the message if nothing else.

Reply:
changed to 

except Exception:
        message = f"Unable to retrieve {key_path} metadata. Got Exception: {Exception}"
        LOGGER.error(message)
        raise Exception(message)

Reply:
I don't think `Got Exception: {Exception}` will return what you want. Are we looking for a callback trace here or a message? In its current form you would just get the object name like `<class 'Exception'>`. I would think you would want something like:

```
except Exception as e:
    message = f"Unable to retrieve {key_path} metadata. Got Exception: {e}"
    LOGGER.error(message, exc_info=True) # If you want tracing https://github.com/nasa/cumulus-message-adapter-python 
    raise Exception from e # Exception chaining https://docs.python.org/3/tutorial/errors.html#exception-chaining
```